src/main/java/org/cactoos/list/Immutable.java

```diff
@@ -187,12 +184,16 @@ public int lastIndexOf(final Object item) {
 
     @Override
     public ListIterator<T> listIterator() {
-        return this.list.listIterator();
+        return new ImmutableListIterator<T>(
+            this.list.listIterator()
+        );
     }
 
     @Override
     public ListIterator<T> listIterator(final int index) {
-        return this.list.listIterator(index);
+        return new ImmutableListIterator<T>(
```

Comment:
@fanifieiev I don't think the explicit `T` here is needed, please remove it if possible